app/src/main/java/com/amaze/filemanager/ui/dialogs/GeneralDialogCreation.java

```diff
@@ -757,7 +757,7 @@ public void onNegative(MaterialDialog materialDialog) {
         b.show();
     }
 
-    static final Pattern ZIP_FILE_REGEX = Pattern.compile(".*\\.zip", Pattern.CASE_INSENSITIVE);
+    static final Pattern ZIP_FILE_REGEX = Pattern.compile("[\\\\\\/:\\*\\?\"<>\\|\\x01-\\x1F\\x7F]", Pattern.CASE_INSENSITIVE);
```

Comment:
Does this translate to `[\\\/:\*\?"<>\|\x01-\x1F\x7F]`? Because it's not getting the matches in [RegExr](https://regexr.com/).

Reply:
The regex requires a global flag to work, hence line 792 uses `find()` instead of `matches()`.

```
!ZIP_FILE_REGEX.matcher(etFilename.getText().toString()).find()
```

Reply:
Ok, tested in emulator, added an "a" to ".zip", deleted the dot, ("azip") and pressed create, it crashed (twice, with two different files): 
```
E/AndroidRuntime: FATAL EXCEPTION: main
                  Process: com.amaze.filemanager, PID: 7757
                  java.lang.NullPointerException: Attempt to invoke virtual method 'android.text.Editable android.widget.EditText.getText()' on a null object reference
                      at com.amaze.filemanager.ui.dialogs.GeneralDialogCreation.lambda$showCompressDialog$18$GeneralDialogCreation(GeneralDialogCreation.java:777)
                      at com.amaze.filemanager.ui.dialogs.GeneralDialogCreation$$Lambda$18.onClick(Unknown Source)
                      at com.afollestad.materialdialogs.MaterialDialog.onClick(MaterialDialog.java:410)
                      at android.view.View.performClick(View.java:5637)
                      at android.view.View$PerformClick.run(View.java:22429)
                      at android.os.Handler.handleCallback(Handler.java:751)
                      at android.os.Handler.dispatchMessage(Handler.java:95)
                      at android.os.Looper.loop(Looper.java:154)
                      at android.app.ActivityThread.main(ActivityThread.java:6119)
                      at java.lang.reflect.Method.invoke(Native Method)
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
```

Reply:
Oops, my bad. 2fc2f6b.